uni/lib/controller/local_storage/database/app_database.dart

```diff
@@ -32,12 +41,28 @@ class AppDatabase {
   /// The version of this database.
   final int version;
 
+  /// Getter to determine if the session is persistent.
+  Future<bool> get persistentSession async {
+    _persistentSession ??=
+        await PreferencesController.getPersistentUserInfo() != null;
+    return _persistentSession!;
+  }
+
   /// Returns an instance of this database.
   Future<Database> getDatabase() async {
     _db ??= await initializeDatabase();
     return _db!;
   }
 
+  Future<void> saveToDatabase(T data);
+
+  /// Calls saveToDatabase if the session is persistent
+  Future<void> saveIfPersistentSession(T data) async {
+    if (await persistentSession) {
+      await saveToDatabase(data);
+    }
```

Comment:
These are the lines that are failing the tests. For some reason, it takes infinite time and the CI fails due to a 10s timeout.

If you comment them, the "Filtered exams" individual test should take about 6 secs.

Reply:
I have made some small fixes in the tests and updated the preferences controller to address this issue.